frontend/lib/framework/error-classes-list.cpp

```diff
@@ -537,4 +537,138 @@ void ErrorProcDefExplicitAnonFormal::write(ErrorWriterBase& wr) const {
   wr.code(fn, {formal});
 }
 
+/* parser errors */
+
+void ErrorParsing::write(ErrorWriterBase& wr) const {
+  auto loc = std::get<const Location>(info);
+  auto errorMessage = std::get<std::string>(info);
+  assert(errorMessage.back() != '.' &&
+         "expected no period at end of ErrorParsing message");
+  errorMessage.push_back('.');
+  wr.heading(kind_, type_, loc, errorMessage);
+}
+
+void ErrorCannotAttachPragmas::write(ErrorWriterBase& wr) const {
+  auto loc = std::get<const Location>(info);
+  wr.heading(kind_, type_, loc, "cannot attach pragmas to this statement.");
```

Comment:
Let's come up with some generic policy for when we want a special error message class over a "general" parser error. The way these are written right now, I don't see much value they have over plain `ErrorParsing`. There are a few things we can do:

1. Print the code. We do have a location, so why not?
2. Add a helpful description of the error message.
   The condition for this particular error message is:
   ```C++
   cs.stmt && cs.stmt->isDecl()
   ```
   This tells me that we can report this.

```suggestion
  wr.heading(kind_, type_, loc, "cannot attach pragmas to this statement.");
  wr.code(loc);
  wr.message("Only declarations, such as variable declarations, can have 'pragma's attached to them.");
```

Reply:
My policy is "complicated or repeated errors", which this one failed to meet before changes.